src/NzbDrone.Common/Http/Dispatchers/ManagedHttpDispatcher.cs

```diff
@@ -1,11 +1,29 @@
 using System;
 using System.Net;
 using NzbDrone.Common.Extensions;
+using com.LandonKey.SocksWebProxy.Proxy;
+using com.LandonKey.SocksWebProxy;
+using System.Net.Sockets;
 
 namespace NzbDrone.Common.Http.Dispatchers
 {
     public class ManagedHttpDispatcher : IHttpDispatcher
     {
+        /// <summary>
+        /// This method returns a free port. Now of course after we find it, it could be opened by something else. But it seems in practice this is not the case.
+        /// 
+        /// The OS will cycle through all free ports randomly in response to this, so its unlikely to happen.
+        /// </summary>
+        /// <returns></returns>
+        private static int GetNextFreePort()
+        {
+            TcpListener l = new TcpListener(IPAddress.Loopback, 0);
```

Comment:
`var` instead of `TcpListener`. Also use a better variable name than `l`, such as `listener`.